src/main/java/org/ontobot/InstanceGenerator.java

```diff
@@ -0,0 +1,91 @@
+package org.ontobot;
+
+import com.google.gson.JsonArray;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonObject;
+import org.semanticweb.owlapi.apibinding.OWLManager;
+import org.semanticweb.owlapi.formats.FunctionalSyntaxDocumentFormat;
+import org.semanticweb.owlapi.model.*;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+
+public class InstanceGenerator {
+
+    private final OWLDataFactory dataFactory;
+    private final OWLOntologyManager manager;
+
+    IRI ontologyIRI;
+
+    OWLOntology ontology;
+
+    public InstanceGenerator(OWLOntology ontology, IRI ontologyIRI) {
+        // Create the OWLOntologyManager and the OWLDataFactory
+        this.manager = OWLManager.createOWLOntologyManager();
+        this.dataFactory = manager.getOWLDataFactory();
+        this.ontology = ontology;
+        this.ontologyIRI = ontologyIRI;
+    }
+
+    public void addInstances(Set<Map.Entry<String, JsonElement>> instanceMsgObjectData) throws FileNotFoundException, OWLOntologyStorageException {
+        OWLNamedIndividual individual = null;
+        OWLDataPropertyAssertionAxiom dataPropAssertion = null;
+        OWLDataProperty dataProp = null;
+        OWLLiteral dataPropLiteral = null;
+        Set<Map.Entry<String, JsonElement>> objectMembersSet = null;
+
+        System.out.println("====================== Instance adder ======================");
+        for (Map.Entry<String, JsonElement> entry : Objects.requireNonNull(instanceMsgObjectData)) {
+            String className = entry.getKey();
+
+            OWLClass owlClass = dataFactory.getOWLClass(IRI.create(ontologyIRI + "#" + className));
+
+            System.out.println(entry.getKey());
+            JsonArray jsonArray = entry.getValue().getAsJsonArray();
+
+            for (JsonElement jsonElement : jsonArray) {
+                JsonObject jsonObject = jsonElement.getAsJsonObject();
+                objectMembersSet = jsonObject.entrySet();
+
+                for (Map.Entry<String, JsonElement> individualsProperty : Objects.requireNonNull((objectMembersSet))) {
+                    if (Objects.equals(individualsProperty.getKey(), "# Object Name")) {
+                        //  Create an individual for the Student class
+                        individual = dataFactory.getOWLNamedIndividual(createIRI("#", individualsProperty.getValue().getAsString()));
+                    } else {
+                        //  Assign data properties to the individual
+                        dataProp = dataFactory.getOWLDataProperty(createIRI("#has", individualsProperty.getKey()));
```

Comment:
individualsProperty.getKey().replace(" ", "_"). In T-box creation, we did this. Otherwise correct data property can't be fetched. remove #has value. because it should be added by the user

Reply:
We have already added data properties while we creating the ontology

Reply:
Already added data properties and newly added data properties should be the same. otherwise, two different data properties will get created in owl file.